perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2627,7 +2657,7 @@ public void updateFormItemData(PerunSession sess, int appId, ApplicationFormItem
 		ApplicationFormItemData existingData = getFormItemDataById(data.getId(), appId);
 		if (existingData == null) throw new RegistrarException("Form item data specified by ID: "+ data.getId() + " not found or doesn't belong to the application "+appId);
 
-		List<Type> notAllowed = Arrays.asList(FROM_FEDERATION_HIDDEN, FROM_FEDERATION_SHOW, USERNAME, PASSWORD, HEADING, HTML_COMMENT, SUBMIT_BUTTON, AUTO_SUBMIT_BUTTON);
+		List<Type> notAllowed = Arrays.asList(FROM_FEDERATION, USERNAME, PASSWORD, HEADING, HTML_COMMENT, SUBMIT_BUTTON, AUTO_SUBMIT_BUTTON);
```

Comment:
This is not correct. Users can change the prefilled value from fed. He cannot change the value only id the field is Hidden/Disabled. Also if it is possible, add some checks to the submitApplication method, so user does not change disabled/hidden value.

Reply:
Fixed.